allennlp/data/token_indexers/openai_transformer_byte_pair_indexer.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, List, Tuple
+import json
+import tarfile
+
+from overrides import overrides
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.file_utils import cached_path
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers.token_indexer import TokenIndexer
+
+
+@TokenIndexer.register("openai_transformer_byte_pair")
+class OpenaiTransformerBytePairIndexer(TokenIndexer[int]):
+    """
+    This is unlike most of our TokenIndexers in that its
+    indexing is not based on a `Vocabulary` but on a fixed
```

Comment:
Could you give a bit more of a description here about this class? It's non-obvious that it would return the ELMo - like mixture of layers, rather than just the last layer as in the original paper. Also maybe add a link to the paper or something.